programs/validator-history/src/instructions/update_priority_fee_history.rs

```diff
@@ -0,0 +1,48 @@
+use crate::{
+    errors::ValidatorHistoryError,
+    state::{Config, ValidatorHistory},
+    utils::cast_epoch,
+};
+use anchor_lang::{prelude::*, solana_program::vote};
+
+#[derive(Accounts)]
+pub struct UpdatePriorityFeeHistory<'info> {
+    #[account(
+      mut,
+      seeds = [ValidatorHistory::SEED, vote_account.key().as_ref()],
+      bump
+  )]
+    pub validator_history_account: AccountLoader<'info, ValidatorHistory>,
+
+    /// CHECK: fine since we are not deserializing account
+    #[account(owner = vote::program::ID.key())]
+    pub vote_account: AccountInfo<'info>,
+
+    #[account(
+      seeds = [Config::SEED],
+      bump = config.bump,
+      has_one = oracle_authority
+  )]
+    pub config: Account<'info, Config>,
+
+    pub oracle_authority: Signer<'info>,
+}
+
+pub fn handle_update_priority_fee_history(
+    ctx: Context<UpdatePriorityFeeHistory>,
+    epoch: u64,
+    lamports: u64,
```

Comment:
Add argument for `priority_fee_tips: 64`, which is the lamports in the PDA during the epoch. This information is trusted from the authority. _Note: The Tip Router program will gather this data from the merkle tree, verify it, and CPI here to upload._